### PR TITLE
Create required group for app examples

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Revert https://github.com/Lightning-AI/lightning/pull/15324/files#r1005673357

### Does your PR introduce any breaking changes? If yes, please list them.

None

cc @carmocca @akihironitta @borda

### CODE REVIEW

**Review by Borda — DISMISSED** [review]
not really, as some examples are used in testing `tests_app`, see for example `pytest_sessionstart` or `test_serve_gradio`

So this would be great to have, but we shall be fully decoupled first :otter:

File: .github/checkgroup.yml [hunk 1]
```diff
@@ -233,27 +233,38 @@ subprojects:
...
-      - "examples/app_*"
+      - "examples/app_*"  # some tests_app tests call examples files
+      - ".github/workflows/ci-app-tests.yml"
+      - "setup.py"
       - ".actions/**"
     checks:
       - "App.cloud-e2e"
       - "make-doctest (app)"
       - "make-html (app)"
-      - "app-examples (macOS-11, app, 3.8, latest)"
```
Comment (Borda) [inline .github/checkgroup.yml:244]: but they shall stay here, as if you change source it may affect examples

Comment (carmocca) [reply .github/checkgroup.yml:244]: They are already in https://github.com/Lightning-AI/lightning/pull/15332/files#diff-5ca41bd8852d2c309825fa76269fb299e7a77b075918a9c56fa1990c40e48f7bR261-R266. If you modify an examples file, that group will cover it

**Review by otaj — APPROVED** [review]

**Review by akihironitta — APPROVED** [review]

File: .github/checkgroup.yml (hunk 1 above)
Comment (otaj) [reply .github/checkgroup.yml:244]: @Borda are we good to resolve so that we can merge this?

**Review by Borda — APPROVED** [review]

**Pull request merged**